File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditDeleteGroupEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.GroupDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDeleteGroupEventDto extends OrganizationEventDto {
    private GroupDto group;

    OrganizationEditDeleteGroupEventDto() {
    }

    public OrganizationEditDeleteGroupEventDto(OrganisationId organisationId, GroupDto group) {
        super(organisationId);
        this.group = group;
    }

    public GroupDto getGroup() {
        return group;
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDeleteGroupEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.GroupDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditDeleteGroupEventDto(
    override val organizationId: OrganizationId,
    val group: GroupDto,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}